ntoskrnl/inbv/inbv.c

```diff
@@ -768,6 +775,7 @@ INIT_FUNCTION
 DisplayBootBitmap(IN BOOLEAN TextMode)
 {
     PVOID Header = NULL, Footer = NULL, Screen = NULL;
+    UCHAR Buffer[128];
```

Comment:
Where does this 128-byte value comes from?

Reply:
Since bar bitmap is 22x9 and 4bpp (hardcoded, yes), minimum value would be `22 * 9 * (4 bpp / 8 bits) = 99`, but I made it a bit larger to reserve.

Also I don't know whether `InbvScreenToBufferBlt` uses some headers for the data.

Reply:
It calls [`VidScreenToBufferBlt`](https://git.reactos.org/?p=reactos.git;a=blob;f=drivers/base/bootvid/i386/vga.c;hb=4f7159e3a51cb155b61fb4ee0949eb9baede2734#l944) (in bootvid.dll).
By looking at the code, this function seems to use `Delta * Height` bytes in the buffer (delta is the last parameter of the function). I don't personally completely understand yet what this `Delta` is for.

Reply:
The scan lines in a bitmap get aligned (to a 4-byte boundary IIRC), so the scanline width can be different from the image width. Delta is likely the scanline width.

Reply:
@yagoulas commented in IRC:

> it looks like delta is the width of the source bitmap
> and I gues this would be used when you want to paint only a part of the marqee when it has reached the right part of the scrollbar and starts disappearing
> this would be a case when width<Delta

Reply:
The hypothesis of the scanline width makes sense, and this would explain the value of "4" seen in other places where this function is called, and where the used Delta is completely different than the specified Width.

Reply:
Ok, so `24 * 9 * (4/8) = 108`, still should be ok.

Reply:
So, `UCHAR Buffer[24 * 9 * 4 / 8];`? ;)

Reply:
Yeah, and it used in three places... :laughing: